react/react1/week3/time-registration/src/SubmitShifts.js

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from "react";
+import "./shifts.css";
+import { ShiftListItem } from "./ShiftOverview";
+
+function FancyBorder(props) {
+  return <div className="fancy-border">{props.children}</div>;
+}
+export function SubmitShifts({ fetchUrl }) {
+  const [list, setList] = useState([]);
+  const [nameInput, setNameInput] = useState("");
+  const [startInput, setStartInput] = useState("");
+  const [endInput, setEndInput] = useState("");
+  useEffect(() => {
+    fetch(fetchUrl)
+      .then((response) => response.json())
+      .then(setList);
+  }, []);
+  const addShift = () => {
+    const newId = new Date().getTime();
+    const newTodo = {
+      name: nameInput,
+      start: startInput,
+      end: endInput,
+      id: newId,
+    };
+    setList((prevTodos) => {
+      return [...prevTodos, newTodo];
+    });
+  };
+  console.log(list);
+  const display = list.map((shift) => (
+    <ShiftListItem
+      name={shift.name}
+      start={shift.start}
+      end={shift.end}
+      key={shift.newId}
+    />
+  ));
+  const searchEmployee = (e) => {
+    const value = e.target.value.toLowerCase();
+    const searchName = list.filter((shift) =>
+      shift.name.toLowerCase().includes(value)
+    );
+    setList(searchName);
+  };
```

Comment:
You search works nicely. The only issue is that once you have searched once, you can't undo that search.

Maybe you can keep a copy of the orignal data you fetched and use that for searching?

Here is an example where I keep a copy of the original data I have fetched. Let me know if it makes sense :)

https://codesandbox.io/s/amazing-diffie-crf0wp?file=/src/App.js